d2bs/kolbot/libs/common/Prototypes.js

```diff
@@ -51,13 +70,116 @@ Party.prototype.__defineGetter__("inTown",
 		return [1, 40, 75, 103, 109].indexOf(this.area) > -1;
 	});
 
-Unit.prototype.__defineGetter__("attacking",
+// Check party member's current act
+Party.prototype.__defineGetter__("act",
+	function () {
+		if (this.area > 0) {
+			if (this.area < 40) {
+				return 1;
+			} else if (this.area < 75) {
+				return 2;
+			} else if (this.area < 103) {
+				return 3;
+			} else if (this.area < 109) {
+				return 4;
+			} else {
+				return 5;
+			}
+		}
+
+		return false;
+	});
+
+// check Difficulty
+Unit.prototype.__defineGetter__("normal",
```

Comment:
don't think lines 94 through 182 make much sense, would rather see comparison with enum types for nightmare/hell/normal and each player type (similar to how NPC is handled)